AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Operations/Helpers/FetchAuthSessionOperationHelper.swift

```diff
@@ -194,4 +203,20 @@ class FetchAuthSessionOperationHelper {
         return session
     }
 
+    private func waitForSignInCancel(authStateMachine: AuthStateMachine) async {
+        let event = AuthenticationEvent(eventType: .cancelSignIn)
+        await authStateMachine.send(event)
+        let stateSequences = await authStateMachine.listen()
+        for await state in stateSequences {
+            guard case .configured(let authenticationState, _) = state else {
+                continue
+            }
+
+            switch authenticationState {
+            case .signedOut:
```

Comment:
do we still get `.signedOut`  even when `signIn` is not done?

Reply:
this logic waits for signedOut state to appear and then return. ie it cancels the signIn flow and put the authstate machine back to signedOut state